marlowe-runtime/web-server/Language/Marlowe/Runtime/Web/Server/HistoryClient.hs

```diff
@@ -208,3 +239,106 @@ loadTransactionsClient eb contractId = MarloweSyncClient $ pure clientInit
             $ SomeTransaction version
             <$> transactions
       }
+
+loadTransactionClient
+  :: EventBackend IO r' HistoryClientSelector
+  -> ContractId
+  -> TxId
+  -> MarloweSyncClient IO (Maybe TxRecord.TxRecord)
+loadTransactionClient eb contractId txId = MarloweSyncClient $ pure clientInit
+  where
+    clientInit = SendMsgFollowContract contractId ClientStFollow
+      { recvMsgContractNotFound =
+          withEvent eb LoadTxNotFound $ const $ pure Nothing
+      , recvMsgContractFound = \block version CreateStep{..} ->
+          withEvent eb LoadTxContractFound \ev -> do
+            addField ev $ show block
+            pure $ clientIdle version $ SeekTx $ pure (block, createOutput)
+      }
+
+    clientIdle :: MarloweVersion v -> SeekTx v -> ClientStIdle v IO (Maybe TxRecord.TxRecord)
+    clientIdle version seekTx = SendMsgRequestNext ClientStNext
+      { recvMsgRollForward = \_ steps ->
+          withEvent eb LoadTxRollForward \ev -> do
+            seekTx' <- foldlM (stepSeekTx ev) seekTx steps
+            pure case seekTx' of
+              Done prevScriptOutputs' tx _ consumingTx -> SendMsgDone
+                $ Just
+                $ mkTxRecord version (snd $ NE.head prevScriptOutputs') tx
+                $ Just consumingTx
+              _ -> clientIdle version seekTx'
+      , recvMsgRollBackward = \block ->
+          withEvent eb LoadTxRollBackward \ev -> do
+            clientIdle version <$> rollbackStepTx ev block seekTx
+      , recvMsgRollBackCreation =
+          withEvent eb LoadTxRollBackCreation $ const $ pure Nothing
+      , recvMsgWait = do
+          withEvent eb LoadTxWait
+            $ const
+            $ pure
+            $ SendMsgCancel
+            $ SendMsgDone
+            $ case seekTx of
+              Done prevScriptOutputs' tx _ consumingTx ->
+                Just $ mkTxRecord version (snd $ NE.head prevScriptOutputs') tx $ Just consumingTx
+              SeekConsuming prevScriptOutputs' tx ->
+                Just $ mkTxRecord version (snd $ NE.head prevScriptOutputs') tx Nothing
+              _ -> Nothing
+      }
+
+    stepSeekTx
+      :: Event IO r HistoryClientSelector LoadTxRollForwardField
+      -> SeekTx v
+      -> ContractStep v
+      -> IO (SeekTx v)
+    stepSeekTx ev seekTx = \case
+      ApplyTransaction newTx -> case seekTx of
+        SeekTx prevOutputs@(prevOutput :| prevOutputs')
+          | txId == transactionId newTx -> do
+              addField ev $ FoundTx $ show txId
+              pure $ SeekConsuming prevOutputs newTx
+          | otherwise -> pure case scriptOutput $ output newTx of
+              Nothing -> seekTx
+              Just scriptOutput -> SeekTx $ (blockHeader newTx, scriptOutput) :| prevOutput : prevOutputs'
```

Comment:
use [`<| :: a -> NonEmpty a -> NonEmpty a`](https://hackage.haskell.org/package/base-4.17.0.0/docs/Data-List-NonEmpty.html#v:-60--124-) instead:
```suggestion
        SeekTx prevOutputs
          | txId == transactionId newTx -> do
              addField ev $ FoundTx $ show txId
              pure $ SeekConsuming prevOutputs newTx
          | otherwise -> pure case scriptOutput $ output newTx of
              Nothing -> seekTx
              Just scriptOutput -> SeekTx $ (blockHeader newTx, scriptOutput) <| prevOutputs
```

Reply:
@jhbertra Don't forget to import `<|` from `Data.List.NonEmpty`